python/nano/src/bigdl/nano/deps/neural_compressor/inc_api_2.py

```diff
@@ -0,0 +1,170 @@
+        if onnxruntime_session_options is None:
+            import onnxruntime
+            onnxruntime_session_options = onnxruntime.SessionOptions()
+        if thread_num is not None:
+            onnxruntime_session_options.intra_op_num_threads = thread_num
+            onnxruntime_session_options.inter_op_num_threads = thread_num
+        if onnx_option == 'tensorflow':
+            from bigdl.nano.deps.onnxruntime.tensorflow.tensorflow_onnxruntime_model import \
+                KerasONNXRuntimeModel
+            return KerasONNXRuntimeModel(q_model.model,
+                                         onnxruntime_session_options=onnxruntime_session_options)
+        else:
+            from bigdl.nano.deps.onnxruntime.pytorch.pytorch_onnxruntime_model import \
+                PytorchONNXRuntimeModel
+            quantized_model = PytorchONNXRuntimeModel(q_model.model, None,
+                                                      onnxruntime_session_options)
+
+            from bigdl.nano.pytorch.utils import patch_attrs_from_model_to_object
+            patch_attrs_from_model_to_object(model, quantized_model)
+            return quantized_model
+
+    else:
+        invalidInputError(False,
+                          f"Invalid framework argument: {not_none_kwargs['framework']}")
+
+
+def _quantize(
+    model,
+    metric,
+    dataloader,
+    framework,
+    conf=None,
+    approach='static',
+    tuning_strategy='bayesian',
+    accuracy_criterion={'relative': 0.99, 'higher_is_better': True},
+    timeout=0,
+    max_trials=1,
+    inputs=[],
+    outpus=[],
+    **kwargs
+):
+    from neural_compressor import quantization
+    from neural_compressor.quantization import PostTrainingQuantConfig
+    from neural_compressor.config import AccuracyCriterion, TuningCriterion
+    from neural_compressor.conf.pythonic_config import Config
+    from neural_compressor.conf.config import QuantConf
+    from neural_compressor.data import DataLoader
+
+    if approach == "dynamic":
+        dataloader = None
+    elif 'onnx' in framework:
+        if hasattr(dataloader, "batch_size"):
+            # `dataloader` is pytorch DataLoader
+            import torch
+
+            def wrap_collate_fn(func):
+                def new_collate_fn(batch):
+                    return [x.numpy() if isinstance(x, torch.Tensor) else x for x in func(batch)]
+                return new_collate_fn
+
+            dataloader = DataLoader(framework, dataloader.dataset,
+                                    collate_fn=wrap_collate_fn(dataloader.collate_fn))
+        else:
+            # `dataloader` is tensorflow (x,y)
+            from .onnx.tensorflow.quantization import KerasNumpyDataset
+            dataloader = KerasNumpyDataset(dataloader[0], dataloader[1], model.dtype)
+    elif not hasattr(dataloader, "batch_size") and not hasattr(dataloader, "_batch_size"):
+        dataloader = DataLoader(framework, dataloader)
```

Comment:
maybe add a comment for this if, otherwise, it's a little hard to know this is for tensorflow's dataset.